controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1520,14 +1513,198 @@ bool master_thread::handle_intel_slave_join(Socket *sd, ieee1905_1::CmduMessageR
+    case 116:
+        return std::vector<uint8_t>{36, 44};
+    case 117:
+        return std::vector<uint8_t>{40, 48};
+    case 118:
+        return std::vector<uint8_t>{52, 56, 60, 64};
+    case 119:
+        return std::vector<uint8_t>{52, 60};
+    case 120:
+        return std::vector<uint8_t>{56, 64};
+    case 121:
+        return std::vector<uint8_t>{100, 104, 108, 112, 116, 120, 124, 128, 132, 136, 140, 144};
+    case 122:
+        return std::vector<uint8_t>{100, 108, 116, 124, 132, 140};
+    case 123:
+        return std::vector<uint8_t>{104, 112, 120, 128, 136, 144};
+    case 124:
+        return std::vector<uint8_t>{149, 153, 157, 161};
+    case 125:
+        return std::vector<uint8_t>{149, 153, 157, 161, 165, 169};
+    case 126:
+        return std::vector<uint8_t>{149, 157};
+    case 127:
+        return std::vector<uint8_t>{153, 161};
+    case 180:
+        return std::vector<uint8_t>{1, 2, 3, 4, 5, 6};
+    default:
+        LOG(ERROR) << "reserved operating class " << operating_class;
+        return std::vector<uint8_t>();
+    }
+}
+
+/**
+ * @brief parse radio capabilities into supported channels which can be saved in the database
+ *        Currently this function only prints the supported operating classes and the non operable
+ *        channel list per each operating class.
+ *        TODO - as part of channel selection task, convert operating classes to supported channels
+ *        so they can be stored in the DB using database.set_hostap_supported_channels()
+ * 
+ * @param radio_caps 
+ * @param supported_channels 
+ * @return true 
+ * @return false 
+ */
+std::vector<beerocks_message::sWifiChannel> master_thread::autoconfig_wsc_parse_radio_caps(
+    std::shared_ptr<wfa_map::tlvApRadioBasicCapabilities> radio_caps)
 {
-    // TODO backhaul handling.
-    // Multi-AP Agent doesn't say anything about the backhaul, so we have to rely on Intel Slave Join for that.
-    std::string backhaul_mac = "";
+    // read all operating class list
+    std::vector<beerocks_message::sWifiChannel> supported_channels;
+    auto operating_classes_list_length = radio_caps->operating_classes_info_list_length();
+    if (operating_classes_list_length > beerocks::message::SUPPORTED_CHANNELS_LENGTH) {
+        LOG(WARNING) << "operating class info list larger then maximum supported channels";
+        operating_classes_list_length = beerocks::message::SUPPORTED_CHANNELS_LENGTH;
+    }
+    for (int oc_idx = 0; oc_idx < operating_classes_list_length; oc_idx++) {
+        std::stringstream ss;
+        auto operating_class_tuple = radio_caps->operating_classes_info_list(oc_idx);
+        if (!std::get<0>(operating_class_tuple)) {
+            LOG(ERROR) << "getting operating class entry has failed!";
+            return supported_channels;
+        }
+        auto &op_class_channels_rx = std::get<1>(operating_class_tuple);
+        auto operating_class       = op_class_channels_rx.operating_class();
```

Comment:
I agree with Moran that it's better to store the operating classes in the database. However, I also agree with Tomer that this is too big a change. So as middle ground, I propose to already add the function to the database to set operating classes, and do the conversion of operating class to channel inside the database.

For this new function, I think it's better to use `add_hostap_supported_operating_class`, not `set_hostap_supported_operating_classes` like is done for the channels (`set_hostap_supported_channels`). That will probably make it easier to separate the parsing from the database function and the database function from how it's internally stored.